src/controls/Widget.ts

```diff
@@ -107,6 +108,54 @@ export class Widget_markdown implements IRequest<'markdown', Widget_markdown_opt
     get result(): Widget_markdown_output { return this.state }
 }
 
+// 🅿️ custom ==============================================================================
+export type Widget_custom_componentProps_ui = {
+    image: (props: {     
+        size?: string | undefined;
+        img: MediaImageID; 
+    }) => JSX.Element
+}
+export type Widget_custom_componentProps<T = unknown> = {
+    value: undefined | T
+    onChange: (value: undefined | T) => void
+    ui: Widget_custom_componentProps_ui
```

Comment:
Oh ! Nice ! if I understand this correctly, this is a dict of widgets you inject so custom components can re-use the existing CushyStudio main components ?

handy !